server/Dockerfile

```diff
@@ -1,25 +1,38 @@
-FROM docker.io/amd64/node:8-stretch
+FROM docker.io/amd64/node:8-stretch AS hyb-node
+
+FROM docker.io/hybsearch/docker-base:v1.2
+
+# The lovely docker community is smart enough to put everything in
+# /usr/local.  We just need to copy (the entire dir) over.
+#
+# NOTE This could probably get tuned up to be more dynamic...
+COPY --from=hyb-node ["/usr/local/", "/usr/local/"]
 
-# enable non-free repos, needed for seq-gen
 RUN sed -i 's|deb http://deb.debian.org/debian stretch main|deb http://http.us.debian.org/debian stretch main non-free|' /etc/apt/sources.list
 
 RUN apt-get update \
-    && apt-get install -y --no-install-recommends \
-        clustalo \
-        mrbayes \
-        seq-gen \
-        python \
-        python-dev \
-        python-pip \
-        python-setuptools \
-    && rm -rf /var/lib/apt/lists/*
+    && apt-get install -qy --no-install-recommends \
+       clustalo \
+       mrbayes \
+       python \
+       python-dev \
+       python-pip \
+       python-setuptools \
+       seq-gen \
+    && apt-get clean autoclean \
+    && rm -rfv /var/lib/{apt,dpkg,cache,log}/
 
-ADD . /hybsearch
+ENV DOCKER=1
+
+RUN mkdir /hybsearch
```

Comment:
I decided to split up the code addition into a few stages.

- Python deps should get manifested and installed first, so just add the requirements.txt file and install them.

- NPM deps get manifested and installed last, because they're most likely to change.

The net result of doing this is that the build is significantly faster. (with caching, that is)